src/BandcampDownloader/Core/DownloadManager.cs

```diff
@@ -465,19 +465,16 @@ internal class DownloadManager {
                     }
                 }
 
-                // Get albums referred on the page
-                regex = new Regex("TralbumData.*\n.*url:.*'/music'\n");
-                if (!regex.IsMatch(htmlCode)) {
-                    // This seem to be a one-album artist with no "music" page => URL redirects to the unique album URL
+                var count = albumsUrls.Count;
+                try {
+                    // We are on a real "music" page if we 
+                    albumsUrls.AddRange(BandcampHelper.GetAlbumsUrl(htmlCode));
+                } catch (NoAlbumFoundException) {
+                    LogAdded(this, new LogArgs($"No referred album could be found on {artistMusicPage}. Try to uncheck the \"Download artist discography\" option", LogType.Error));
+                }
+                // This seem to be a one-album artist with no "music" page => URL redirects to the unique album URL
+                if(albumsUrls.Count - count == 0) {
                     albumsUrls.Add(url);
```

Comment:
This seem to be making the following assumption: if the call to `albumsUrls.AddRange(BandcampHelper.GetAlbumsUrl(htmlCode));` raised an exception, that means that we're on a "one-album artist with no "music" page"; but it's not necessarily true. The previous version of the code shows that we could be on a real "music" page, but the call could raise an exception for any number of reason. The issue with the following code is that we could call `albumsUrls.Add(url);` whereas `url` is not an album URL.

I think we should find a better way to know if we're on a music page or not.

Reply:
I haven't studied the the full download flow, but my assumption was not relying on some html regex matching makes the code less fragile wrt to bandcamp backend logic changes. 

Feel free to close this/suggest a better way - the reason I've made the PR is it's more efficient to optimistically assume the fix that worked for me would be fine, so I didn't create issue w/ any discussion etc.

Reply:
>not relying on some html regex matching makes the code less fragile wrt to bandcamp backend logic changes

You are right. I'm going to merge this and build on it.